examples/semantic/src/main/scala/289.scala

```diff
@@ -0,0 +1,13 @@
+package ticket289
+
+case class C[T: List](x: Int)
+
+case class C2[T: List: Option](x: Int)(y: String)
```

Comment:
Where can I read about what `T: List: Option` means? Some inspection in the Scala REPL indicates this is an evidence parameter that must be supplied implicitly.

Reply:
Ok, I understand this now. It's an application of the Curry-Howard correspondence between types and propositions. `[T: List: Option]` treats `List` and `Option` as propositions on `T`, and desugars to implicit evidence parameters `(implicit evidence$1: List[T], evidence$2: Option[T])`. The constructor can only be called in contexts where we have evidence that types `List[T]` and `Option[T]` are inhabited, which is equivalent to saying that we have a proof that the propositions `List[T]` and `Option[T]` are true.

Source: https://stackoverflow.com/questions/34499663/what-precisely-is-a-scala-evidence-parameter

Even though this is "just a test input," it definitely deserves a comment explaining what this is. Maybe also include the desugared version directly in the test input, with a comment above it saying "// The above desugars to this:"

I see "Context Bounds and View Bounds" of https://www.scala-lang.org/files/archive/spec/2.11/07-implicits.html also mentions the bounds `‘>:’ Type`, `‘<:’ Type`, and `‘<%’ Type`. Might as well test for these as well.

Reply:
I feel that context bounds are well-known enough in the scala community that such a desugaring comment is actually unnecessary. Also, the ticket as referenced by the file is https://github.com/twitter/rsc/issues/289, which is specifically for context bounds. View bounds are quite deprecated and I've yet to see any in practice. Type bounds don't produce any desugarings. however, those are fairly easy to include in a test so will do.